docs/ext/ipc.md

```diff
@@ -0,0 +1,200 @@
+# IPC
+
+This is a description of the **Common IPC Metrics** that can be published by various IPC
+libraries, with the goal of allowing consolidated monitoring and analysis across differing
+IPC implementations.
+
+## Dimensions Common to All Metrics
+
+Not all dimensions are applicable for all of the metrics, and later in the sections 
+for each specific metric, the applicable dimensions are specified.
+
+Also note that not all dimensions have been implemented or are applicable for _all_
+implementations.
+
+* `ipc.protocol`: A short name of the network protocol in use, eg. `grpc`, `http_1`,
+  `http_2`, `udp`, etc ...
+* `ipc.vip`: The Eureka VIP address used to find the the server.
+* `ipc.result`: Was this considered by the implementation to be successful. Allowed Values =
+  [`success`, `failure`].
+* `ipc.status`: One of a predefined list of status values indicating the general result, eg.
+  success, bad_request, timeout, etc… See the _ipc.status values section below_.
+* `ipc.status.detail`: For cases where the ipc.status needs to be further subdivided, this tag
+  can hold an additional more specific detail, likely ipc-implementation specific. eg status of
+  connection_error and detail of no_servers / connect_timeout / ssl_handshake_failure.
+* `ipc.failure.injected`: Indicates that an artificial failure was injected into the request
+  processing for testing purposes. The outcome of that failure will be reflected in the other
+  error tags. Allowed Values = [true]
+* `ipc.endpoint`: The name of the endpoint/function/feature the message was sent to within
+  the server (eg. the URL path prefix for a java servlet, or the grpc endpoint name).
+* `ipc.attempt`: Which attempt at sending this message is this. Allowed Values =
+  [`initial`, `second`, `third_up`] (`initial` is the first attempt, `second` is 2nd attempt
+  but first *retry*, `third_up` means third or higher attempt).
+* `ipc.attempt.final`: Indicates if this request was the final attempt of potentially multiple
+  retry attempts. Allowed Values = [`true`, `false`].
+* `ipc.server.app`: The `nf.app` of the server the message is being sent *to*.
+* `ipc.server.asg`: The `nf.asg` of the server the message is being sent *to*.
+* `ipc.client.app`: The `nf.app` of the server the message is being sent *from*.
+* `ipc.client.asg`: The `nf.asg` of the server the message is being sent *from*.
```

Comment:
Were `ipc.server.cluster` and `ipc.client.cluster` omitted on purpose?

Reply:
They were, although maybe they shouldn't be. It's because I omitted them in the original proposal doc (with the thinking being that the minimum of `nf.asg` was needed), but then I ended up adding cluster in the implementations I worked on ...  so maybe the spec should follow the implementations here and we should document these?

Reply:
I think they should be included in the docs if they will show up on the metrics.